mungegithub/mungers/publish_scripts/publish_client_go.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script publishes the latest changes in the ${src_branch} of
+# k8s.io/kubernetes/staging/src/client-go to the ${dst_branch} of
+# k8s.io/client-go.
+#
+# The script assumes that the working directory is $GOPATH/src/k8s.io/client-go.
+# It also assumes $GOPATH/k8s.io/apimachinery has been updated.
+#
+# The script is expected to be run by
+# k8s.io/test-infra/mungegithub/mungers/publisher.go
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+if [ ! $# -eq 3 ]; then
+    echo "usage: $0 token src_branch dst_branch"
+    exit 1
+fi
+
+TOKEN="${1}"
+# src branch of k8s.io/kubernetes
+SRC_BRANCH="${2:-master}"
+# dst branch of k8s.io/client-go
+DST_BRANCH="${3:-master}"
+readonly TOKEN SRC_BRANCH DST_BRANCH
+
+SCRIPT_DIR=$(dirname "${BASH_SOURCE}")
+
+git checkout "${DST_BRANCH}"
+
+source "${SCRIPT_DIR}"/util.sh
+
+set_github_token "${TOKEN}"
+trap cleanup_github_token EXIT SIGINT
+
+# this currently only updates commit hash of k8s.io/apimachinery
+update_godeps_json() {
+    godeps_json="./Godeps/Godeps.json"
+    local old_revs
+    local new_rev=$(cd ../apimachinery; git rev-parse HEAD)
+
+    while read path rev; do
+        if [[ "${path}" == "k8s.io/apimachinery"* ]]; then
+            old_revs+="${rev}"$'\n'
+        fi
+    done < <(jq '.Deps|.[]|.ImportPath + " " + .Rev' -r < "${godeps_json}")
+    old_revs=$(echo "${old_revs%%$'\n'}" | sort | uniq)
+    while read old_rev; do
+        sed -i "s|${old_rev}|${new_rev}|g" "${godeps_json}"
+    done <<< "${old_revs}"
```

Comment:
There is certainly an elegant jq command replacing these 9 lines. I wish I was a better jq wizard.....